assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Ads Settings Section.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ActionCard, Grid, Button, TextControl, CheckboxControl, SelectControl } from '../';
+
+const SettingsSection = ( { settings, disabled, onChange, onUpdate } ) => {
+	const section = settings.find( setting => ! setting.key || setting.key === 'active' );
+	if ( ! section ) {
+		return null;
+	}
+	const activation = settings.find( setting => setting.key === 'active' );
+	const fields = settings.filter( setting => setting.key && setting.key !== 'active' );
+	const getControlComponent = setting => {
```

Comment:
No need for this function to be defined in the component – can be moved out for better readability & performance.